kernel/packages/entryPoints/preview.ts

```diff
@@ -25,12 +25,19 @@ if (!container) throw new Error('cannot find element #gameContainer')
 
 const defaultScene: IFuture<ILand> = future()
 
+let wsScene: string | undefined = undefined
+
+if (location.search.indexOf("WS_SCENE") !== -1) {
+  wsScene = `${location.protocol === 'https:' ? 'wss' : 'ws'}://${document.location.host}/?scene`
+}
+
 function startPreviewWatcher() {
   // this is set to avoid double loading scenes due queued messages
   let isSceneLoading: boolean = true
 
   const loadScene = () => {
-    loadPreviewScene()
+    isSceneLoading = true
```

Comment:
Why was this change necessary? How does this work now?

Reply:
Setting the flag was moved inside the function.

Before it was outside, right before of the function call, a dozen lines below.

```diff
-isSceneLoading = true
 loadScene()
```